### ISSUE TITLE
panic: non-positive interval for NewTicker on 1.10.1 after upgrade

### ISSUE THREAD

**Issue by arren-ru** [hashicorp/vault#15147]
**Describe the bug**
After upgrading from 1.10.0 to 1.10.1 unable to start vault, after unsealing it crashes with Go panic.

**Environment:**
* Vault Server and CLI Versions are 1.10.1
* Server Operating System/Architecture: archlinux with systemd

Vault server configuration file(s):
```hcl
ui = true
disable_mlock = true

storage "raft" {
  path = "/var/lib/vault"
}

api_addr = "https://127.0.0.1:8200"
cluster_addr = "https://127.0.0.1:8201"

listener "tcp" {
  address       = ":8200"
  tls_cert_file = "***.crt"
  tls_key_file  = "***.key"
}

service_registration "consul" {
  address         = "127.0.0.1:8501"
  service_address = "10.10.0.10"
  scheme          = "https"
  token           = "***"
}
```

Crash logs are:
```
Apr 24 01:12:26 helios vault[696416]: 2022-04-24T01:12:26.899+0700 [INFO]  core: starting raft active node
Apr 24 01:12:26 helios vault[696416]: 2022-04-24T01:12:26.899+0700 [INFO]  storage.raft: starting autopilot: config="&{false 0 10s 24h0m0s 1000 0 10s}" reconcile_interval=0s
Apr 24 01:12:26 helios vault[696416]: panic: non-positive interval for NewTicker
Apr 24 01:12:26 helios vault[696416]: goroutine 2346 [running]:
Apr 24 01:12:26 helios vault[696416]: time.NewTicker(0xdf8475800?)
Apr 24 01:12:26 helios vault[696416]:         time/tick.go:24 +0x10f
Apr 24 01:12:26 helios vault[696416]: github.com/hashicorp/vault/vault.(*Core).raftTLSRotatePhased.func3()
Apr 24 01:12:26 helios vault[696416]:         github.com/hashicorp/vault/vault/raft.go:486 +0x10a
Apr 24 01:12:26 helios vault[696416]: created by github.com/hashicorp/vault/vault.(*Core).raftTLSRotatePhased
Apr 24 01:12:26 helios vault[696416]:         github.com/hashicorp/vault/vault/raft.go:477 +0x42a
Apr 24 01:12:26 helios systemd[1]: vault.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Apr 24 01:12:26 helios systemd[1]: vault.service: Failed with result 'exit-code'.
```

Additional note is that initializing new instance is ok and works fine, but in case of upgrading I don't want to loose data.
Version 1.10.0 works fine without any error.

**Comment by sjm42** [comment]
Well, I can confirm the same issue.
Environment: Ubuntu 20.04 LTS on 64bit ARM/aarch64 on aws (t4g.small)
Vault binary is installed from official deb package built & published by hashicorp.

This "cluster" has only one node, if that matters.

Server config:

```
storage "raft" {
  path    = "/vault/i-XXXX"
  node_id = "i-XXXX"
  retry_join {
      leader_api_addr = "https://10.XXX.YY.ZZZ:8200"
  }
}

listener "tcp" {
  address     = "0.0.0.0:8200"
  cluster_address     = "0.0.0.0:8201"
  tls_cert_file = "/vault/tls/vault-i.pem"
  tls_key_file  = "/vault/tls/vault-i.key"
}

seal "awskms" {
  region     = "eu-central-1"
  kms_key_id = "XXX-YYY-ZZZ"

}

api_addr = "https://10.XXX.YY.ZZZ:8200"
cluster_addr = "https://10.XXX.YY.ZZZ:8201"
disable_mlock = true
ui = true
```

crash log:

```
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.215Z [INFO]  core: acquired lock, enabling active operation
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.237Z [INFO]  core: post-unseal setup starting
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.240Z [INFO]  core: loaded wrapping token key
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.240Z [INFO]  core: successfully setup plugin catalog: plugin-directory=""
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.241Z [INFO]  core: successfully mounted backend: type=system path=sys/
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.241Z [INFO]  core: successfully mounted backend: type=identity path=identity/
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.242Z [INFO]  core: successfully mounted backend: type=kv path=kv/
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.242Z [INFO]  core: successfully mounted backend: type=pki path=pki/
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.242Z [INFO]  core: successfully mounted backend: type=cubbyhole path=cubbyhole/
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.244Z [INFO]  core: successfully enabled credential backend: type=token path=token/
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.244Z [INFO]  rollback: starting rollback manager
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.244Z [INFO]  core: restoring leases
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.244Z [INFO]  identity: entities restored
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.245Z [INFO]  identity: groups restored
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.245Z [INFO]  core: starting raft active node
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.245Z [INFO]  storage.raft: starting autopilot: config="&{false 0 10s 24h0m0s 1000 0 10s}" reconcile_interval=0s
Apr 25 09:11:01 vault-i vault[4935]: 2022-04-25T09:11:01.244Z [INFO]  expiration: lease restore complete
Apr 25 09:11:01 vault-i vault[4935]: panic: non-positive interval for NewTicker
Apr 25 09:11:01 vault-i vault[4935]: goroutine 434 [running]:
Apr 25 09:11:01 vault-i vault[4935]: time.NewTicker(0xfff82a6cbcb9a9f9)
Apr 25 09:11:01 vault-i vault[4935]: 	/opt/hostedtoolcache/go/1.17.9/x64/src/time/tick.go:24 +0x14c
Apr 25 09:11:01 vault-i vault[4935]: github.com/hashicorp/vault/vault.(*Core).raftTLSRotatePhased.func3(0x400037ad20, 0x4000b9c300, {0x5c456a0, 0x400061bf20}, 0x4000b9c2c0, 0x40006d1d40)
Apr 25 09:11:01 vault-i vault[4935]: 	/home/runner/work/vault/vault/vault/raft.go:486 +0xe8
Apr 25 09:11:01 vault-i vault[4935]: created by github.com/hashicorp/vault/vault.(*Core).raftTLSRotatePhased
Apr 25 09:11:01 vault-i vault[4935]: 	/home/runner/work/vault/vault/vault/raft.go:477 +0x344
```

The strange thing is, meanwhile I am running a 3-node cluster on practically identical setup, same vault version 1.10.1 on arm64 with raft backend with no issues. The said cluster has been very inactive, though.

**Comment by localhorst-org** [comment]
Same for me,

Ubuntu 20.04, vault installation from official deb-repo

deb [arch=amd64] https://apt.releases.hashicorp.com focal main

update vault from 1.10.0 to 1.10.1

i'am also on the way with a "raft storage single node" installation. Saw in the logs, after unseal

[...]
Apr 25 09:05:10 auth01 vault[3692820]: panic: non-positive interval for NewTicker
Apr 25 09:05:10 auth01 vault[3692820]: goroutine 492 [running]:
Apr 25 09:05:10 auth01 vault[3692820]: time.NewTicker(0xdf8475800)
Apr 25 09:05:10 auth01 vault[3692820]: #011/opt/hostedtoolcache/go/1.17.9/x64/src/time/tick.go:24 +0x10f
Apr 25 09:05:10 auth01 vault[3692820]: github.com/hashicorp/vault/vault.(*Core).raftTLSRotatePhased.func3()
Apr 25 09:05:10 auth01 vault[3692820]: #011/home/runner/work/vault/vault/vault/raft.go:486 +0x10a
Apr 25 09:05:10 auth01 vault[3692820]: created by github.com/hashicorp/vault/vault.(*Core).raftTLSRotatePhased
Apr 25 09:05:10 auth01 vault[3692820]: #011/home/runner/work/vault/vault/vault/raft.go:477 +0x3e5
Apr 25 09:05:11 auth01 systemd[1]: vault-dev.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Apr 25 09:05:11 auth01 systemd[1]: vault-dev.service: Failed with result 'exit-code'.
Apr 25 09:05:16 auth01 systemd[1]: vault-dev.service: Scheduled restart job, restart counter is at 2.
[...]

- - - - - 

in a dev environent within docker container, container from official docker hub -> https://hub.docker.com/_/vault/ and also single node raft storage.

Update with "docker pull vault:latest"   from 1.10.0 to 1.10.1

panic: non-positive interval for NewTicker

goroutine 1047 [running]:
time.NewTicker(0xdf8475800)
        /opt/hostedtoolcache/go/1.17.9/x64/src/time/tick.go:24 +0x10f
github.com/hashicorp/vault/vault.(*Core).raftTLSRotatePhased.func3()
        /home/runner/work/vault/vault/vault/raft.go:486 +0x10a
created by github.com/hashicorp/vault/vault.(*Core).raftTLSRotatePhased
        /home/runner/work/vault/vault/vault/raft.go:477 +0x3e5


In both cases downgrade to the prior Version 1.10.0 brings me back on track

**Comment by ccapurso** [comment]
Thank you for bringing this to our attention, @arren-ru. We appreciate everyone's crash logs as they have helped identify the underlying issue. Our team has begun looking into this and are working to provide a fix. I will keep you posted on when it will be made available.

**Comment by tristanmorgan** [comment]
Before the upgrade I think the new Raft TLS keys are not being marked as complete when used on a single node. as Evidence of messages like this:
```
2022-04-26T11:20:23.456+1000 [INFO]  core: starting raft active node
2022-04-26T11:20:23.457+1000 [INFO]  storage.raft: starting autopilot: config="&{false 0 10s 24h0m0s 1000 0 10s}" reconcile_interval=0s
2022-04-26T11:20:23.457+1000 [WARN]  core.raft: skipping new raft TLS config creation, keys are pending
2022-04-26T11:20:23.561+1000 [INFO]  core: post-unseal setup complete
2022-04-26T11:25:23.456+1000 [WARN]  core.raft: skipping new raft TLS config creation, keys are pending
```

**Comment by marceloavilaoliveira** [comment]
Hello, is there a workaround to this problem? Is the 1.10.1 version unusable, and we must wait for 1.10.2?

**Comment by hsimon-hashicorp** [comment]
> Hello, is there a workaround to this problem? Is the 1.10.1 version unusable, and we must wait for 1.10.2?

At this time, our best guidance is to downgrade to 1.10.0, unfortunately.

**Comment by marceloavilaoliveira** [comment]
No problem at all, it would be good to place a message in the 1.10.1 release notes. Thanks a lot for your prompt answer.

**Comment by ccapurso** [comment]
>  it would be good to place a message in the 1.10.1 release notes

We have added this issue to the known list of issues for the [1.10.1 release](https://www.vaultproject.io/docs/upgrading/upgrade-to-1.10.x#integrated-storage-panic-related-to-old-tls-key) in our upgrade docs. Please keep in mind that this affects more than just 1.10.1.

Per the upgrade notice:

> Impacted versions: 1.10.1, 1.9.5, 1.8.10. Versions prior to these are unaffected.

With that said, it is recommended to not perform minor upgrades past 1.10.0, 1.9.4, and 1.8.9 at this time. As @hsimon-hashicorp has mentioned, we are working to get the fix in https://github.com/hashicorp/vault/pull/15156 released for versions 1.8.11, 1.9.6, and 1.10.2 as soon as possible. I will update this issue when they are available. Thank you for your patience and understanding!

**Comment by marceloavilaoliveira** [comment]
Thank you very much for your detailed response.

**Comment by ccapurso** [comment]
Thank you for your patience, everyone! We have released a fix for this issue as part of 1.8.11, 1.9.6, and 1.10.2. Below are a list of links where you can find the associated binaries and docker images. Let me know if you have any questions!

**Open Source Binaries**
- [1.8.11](https://releases.hashicorp.com/vault/1.8.11/)
- [1.9.6](https://releases.hashicorp.com/vault/1.9.6/)
- [1.10.2](https://releases.hashicorp.com/vault/1.10.2/)

**Enterprise Binaries**
- [1.8.11+ent](https://releases.hashicorp.com/vault/1.8.11+ent/)
- [1.9.6+ent](https://releases.hashicorp.com/vault/1.9.6+ent/)
- [1.10.2+ent](https://releases.hashicorp.com/vault/1.10.2+ent/)

**Open Source Docker Images**
- [1.8.11](https://hub.docker.com/layers/vault/hashicorp/vault/1.8.11/images/sha256-1fa8600d5467f02163fa8b0562723c11002104480faad13e4313d4a7c5abdfe8?context=explore)
- [1.9.6](https://hub.docker.com/layers/vault/hashicorp/vault/1.9.6/images/sha256-9f3af18214e9e9a1d7d4ecb7890be10d4fe0873bf6a1a213d111f0c9e4e22a31?context=explore)
- [1.10.2](https://hub.docker.com/layers/vault/hashicorp/vault/1.10.2/images/sha256-9e6a6ccfa41e7fe93e7b5423f5042d06545d4de24a0a7a42a9d4e1f1d9654de5?context=explore)

**Enterprise Docker Images**
- [1.8.11-ent](https://hub.docker.com/layers/vault-enterprise/hashicorp/vault-enterprise/1.8.11-ent/images/sha256-54a3a9358243504646264481f6e98973da8edb7f36ce0d02b5468fbf98d32878?context=explore)
- [1.9.6-ent](https://hub.docker.com/layers/vault-enterprise/hashicorp/vault-enterprise/1.9.6-ent/images/sha256-151534703681602a11c156583ffba9c983df40bfd8bfbef70d1f027351d97b5c?context=explore)
- [1.10.2-ent](https://hub.docker.com/layers/vault-enterprise/hashicorp/vault-enterprise/1.10.2-ent/images/sha256-555e9b67c050c4b307413d174ce38d119bb313f41a5d4b62cfdc40e5d4026a63?context=explore)
